clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -4020,6 +4020,39 @@ def FMinOp : BinaryFPToFPBuiltinOp<"fmin", "MinNumOp">;
 def FModOp : BinaryFPToFPBuiltinOp<"fmod", "FRemOp">;
 def PowOp : BinaryFPToFPBuiltinOp<"pow", "PowOp">;
 
+def IsFPClassOp : CIR_Op<"is_fp_class"> {
+  let summary = "Corresponding to the `__builtin_fpclassify` builtin function in clang";
+
+  let description = [{
+    The `cir.is_fp_class` operation takes a floating-point value as its first
+    argument and a bitfield of flags as its second argument. The operation
+    returns a boolean value indicating whether the floating-point value
+    satisfies the given flags.
+
+    The flags must be a compile time constant and the values are:
+
+    | Bit # | floating-point class |
+    | -------- | ------- |
+    | 0 | Signaling NaN      |
+    | 1 | Quiet NaN          |
+    | 2 | Negative infinity  |
+    | 3 | Negative normal    |
+    | 4 | Negative subnormal |
+    | 5 | Negative zero      |
+    | 6 | Positive zero      |
+    | 7 | Positive subnormal |
+    | 8 | Positive normal    |
+    | 9 | Positive infinity  |
+  }];
+
+  let arguments = (ins CIR_AnyFloat:$src,
+                       I32Attr:$flags);
+  let results = (outs CIR_BoolType:$result);
+  let assemblyFormat = [{
+    `(` $src`,` $flags `)` `:` functional-type($src, $result) attr-dict
```

Comment:
style nit: we prefer operand list without parenthesis:

```suggestion
    $src `,` $flags `:` functional-type($src, $result) attr-dict
```

@bcardosolopes The current style guide says we should exclude the parenthesis around the input types if there is only one input operand. But `functional-type` adds parenthesis to input type list without conditions, so maybe we should update the assembly style guide accordingly?

Reply:
Although $flags are not attribtue, but I think it is an input too. It looks odder to remove the parenthesis.

Reply:
Maybe we need to refine the style guide.

Reply:
I think it's fine if function-type does, but my take is that we should avoid, any improvement to the docs are more than welcome. `cir.vtt.address_point` has more than one param and seems nice.

Reply:
@bcardosolopes  thanks. Just to make sure I read correctly, does the current `assemblyFormat ` looks good to you?

Reply:
Ready to land this once the parens are removed (https://llvm.github.io/clangir/Dialect/cir-style-guide.html#arguments-list), if you feel uncomfortable with the the number being alone, perhaps we could have:
```
cir.is_fp_class %1, flags = 504 : (!cir.float) -> !cir.bool
```

Reply:
Done. It looks like I can't print `flags` due to rules. But it is not so bad to drop it.